bin/generateBibfile.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+
+r"""Utility that interogates ook (lsst.io) to get doc metadata
+then generates a bibfile fpr inclusion in texmf.
+"""
+
+import argparse
+from algoliasearch.search_client import SearchClient
+from datetime import datetime
+import sys
+import calendar
+
+MAXREC = 5000
+
+
+def write_latex_bibentry(auth, title, year, month, handle, fd=sys.stdout):
+    """ Write a bibentry for document info passed
+    Parameters
+    ----------
+    auth : `str``
+        Author list ready for tex
+    title : `str`
+    year  : `str`
+    month  : `str`
+    handle  : `str`
+
+    """
+
+    print("@DocuShare{{{},".format(handle), file=fd)
+    print("   author = {{{}}},".format(auth), file=fd)
+    print("    title = {{{}}},".format(title), file=fd)
+    print("     year = {},".format(year), file=fd)
+    print("    month = {},".format(month), file=fd)
+    print("   handle = {{{}}},".format(handle), file=fd)
+    print("      url = {{http://{}.lsst.io }} }}".format(handle), file=fd)
+
+
+def generate_bibfile(outfile, query):
+
+    client = SearchClient.create('0OJETYIVL5', 'b7bd2f1080a5c4fe5eee502462bcc9d3')
+    index = client.init_index('document_dev')
+
+    params = {
+        'attributesToRetrieve': [
+            'handle',
+            'series',
+            'h1',
+            'baseUrl',
+            'sourceUpdateTime',
+            'sourceUpdateTimestamp',
+            'authorNames'
+        ],
+        'hitsPerPage': MAXREC
+    }
+
+    res = index.search(query, params)
+
+    print('## DO NOT EDIT THIS FILE. It is generated from generateBibfile.py\n'
+          '## Add static entries in etc/static_entries.bib (or remove them if they clash.\n'
+          '## This files should contain ALL entries on www.lsst.io ', file=outfile)
+
+    bcount = 0
+    for count, d in enumerate(res['hits']):
+        if d['series'] == "TESTN":
```

Comment:
I'd suggest also testing if series is in the hit result. At some point we'll start adding records that don't have handles and those will need to treated differently:

```suggestion
        if 'series' in d.keys() or d['series'] == 'TESTN':
```

For extra robustness, you might also want this script to opt into specific series rather than only excluding specific series. i.e. `d['series'] not in allowed_series` where `allowed_series` is a list of known document handles.

Reply:
i want them all only TESTN  do not form good entries ..  i have taken that bit with an AND nor an OR .. the or would still check the second term even if the first failed.